### PR TITLE
Implement binary distribution for lmstudio-tray-manager

### PR DESCRIPTION
**Pull request by Ajimaru**
## Summary

This PR merges the `dev` branch into `main` with the following code quality and maintenance improvements:

### Changes Made

1. **Consolidated Pyre ignore entries** (.gitignore):
   - Removed redundant `.pyre_check/`, `.pyre_configuration/`, and `.pyre_configuration.local/` entries
   - Kept only `.pyre/` which covers all subdirectories

2. **Simplified test docstrings** (tests/test_lmstudio_tray.py):
   - Replaced verbose multi-line docstrings with concise one-liners
   - Updated `test_load_version_from_dir_empty_file` and `test_version_flag_exits` for consistency

3. **Improved function docstrings** (lmstudio_tray.py):
   - Updated `parse_args()` to follow Google-style guidelines by removing empty "Args: None" and "Raises: None" sections
   - Updated `load_version_from_dir()` to remove unconventional "Raises:" section documenting non-raising exceptions

4. **Enhanced subprocess timeout handling** (build_binary.py):
   - Added 3600-second timeout to PyInstaller subprocess.run() call
   - Wrapped call in try/except to catch and handle subprocess.TimeoutExpired
   - Prevents build process from hanging indefinitely

5. **Improved command printing** (build_binary.py):
   - Added `import shlex`
   - Replaced `" ".join(cmd)` with `shlex.join(cmd)` for proper shell argument quoting

6. **Fixed GdkPixbuf loader bundling** (lmstudio-tray-manager.spec):
   - Added `import glob`
   - Changed from appending directory to iterating through individual `.so*` files
   - Each loader file is now added as an explicit tuple to binaries list
   - Ensures PyInstaller receives file paths instead of bare directories

7. **Added logging for build diagnostic** (lmstudio-tray-manager.spec):
   - Added `import logging`
   - Replaced silent except clause with `logging.warning()` that includes exception details
   - Better debugging when pkg-config/GdkPixbuf detection fails

### Quality Checks

All changes have been verified and validated with Codacy analysis to ensure no new issues were introduced.

### Testing

Tests pass successfully (exit code 0).

___

### **PR Type**
Enhancement, Tests

___

### **Description**
- Add PyInstaller binary build system with build.sh and build_binary.py

- Implement argparse-based CLI with short flags (-d, -a, -g, -v) for tray script

- Add auto-start daemon and GUI launch modes via command-line flags

- Expand test coverage with 50+ new tests for edge cases and workflows

- Update autostart script to prefer binary when available

- Enhance documentation with binary release guide and updated workflows

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["lmstudio_tray.py"] -->|"argparse CLI"| B["parse_args()"]
  B -->|"--auto-start-daemon"| C["_maybe_auto_start_daemon()"]
  B -->|"--gui"| D["_maybe_start_gui()"]
  B -->|"--version"| E["load_version_from_dir()"]
  F["build_binary.py"] -->|"PyInstaller"| G["dist/lmstudio-tray-manager"]
  H["build.sh"] -->|"orchestrates"| F
  I["lmstudio_autostart.sh"] -->|"prefers binary"| G
  I -->|"fallback"| A
  J["tests/test_build_binary.py"] -->|"validates"| F
  K["tests/test_lmstudio_tray.py"] -->|"50+ new tests"| A
```

<details><summary><h3>File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>4 files</summary><table>
<tr>
  <td><strong>lmstudio_tray.py</strong><dd><code>Add argparse CLI and auto-start modes</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/Ajimaru/LM-Studio-Tray-Manager/pull/20/files#diff-2a1e6ec53f290163bda79a84ffe0fe34a121b5d5eda1bc81307249629989aede">+147/-31</a></td>

</tr>

<tr>
  <td><strong>build_binary.py</strong><dd><code>New PyInstaller build script with GTK bundling</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/Ajimaru/LM-Studio-Tray-Manager/pull/20/files#diff-da524db6fe8040689c21d15fefe6256f16cf263e57eafb4bae3116a3eaf28981">+204/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>build.sh</strong><dd><code>Automated binary build with optimization</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/Ajimaru/LM-Studio-Tray-Manager/pull/20/files#diff-4d2a8eefdf2a9783512a35da4dc7676a66404b6f3826a8af9aad038722da6823">+152/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>lmstudio_autostart.sh</strong><dd><code>Prefer binary when available, forward flags</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/Ajimaru/LM-Studio-Tray-Manager/pull/20/files#diff-2f34389ebdeaba6a5bbd4156af993e897ba68b5dd67ecc19ab70cf4276c66f1f">+31/-10</a>&nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>4 files</summary><table>
<tr>
  <td><strong>lmstudio-tray-manager.spec</strong><dd><code>PyInstaller spec file for binary packaging</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/Ajimaru/LM-Studio-Tray-Manager/pull/20/files#diff-8bb7819c75c70c60c05d6470938df7da88f9b69c52978ce6bf7bb4b3f6435b38">+87/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>release.yml</strong><dd><code>Add binary build and dual-package release</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/Ajimaru/LM-Studio-Tray-Manager/pull/20/files#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34">+70/-16</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>pytest.ini</strong><dd><code>Include build_binary in coverage</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/Ajimaru/LM-Studio-Tray-Manager/pull/20/files#diff-fb6a686182f16eb54af3c628f38593f347f68aba31de903983023c560288d7a1">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>VERSION</strong><dd><code>Bump version to v0.3.0</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/Ajimaru/LM-Studio-Tray-Manager/pull/20/files#diff-7b60b8e351cbb80c47459ffe2c79f1a26404871f49294780fe47ad0e58c09350">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Tests</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td><strong>test_lmstudio_tray.py</strong><dd><code>Add 50+ new tests for CLI and workflows</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/Ajimaru/LM-Studio-Tray-Manager/pull/20/files#diff-ac1941213ca83c520fc521bff598459faad7fd7c997a1b6b2d19b691a01fb0d1">+574/-4</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>test_build_binary.py</strong><dd><code>New test suite for build_binary.py</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/Ajimaru/LM-Studio-Tray-Manager/pull/20/files#diff-72321ba573c5a2578947ab976da972aaf2730a1556186d9b073b435707a7cef5">+219/-0</a>&nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Documentation</strong></td><td><details><summary>8 files</summary><table>
<tr>
  <td><strong>guide.html</strong><dd><code>Add binary release section and flow diagrams</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/Ajimaru/LM-Studio-Tray-Manager/pull/20/files#diff-ad2c5b621b06ac143f5dd465eee80172c6842ead7dca481a0cc1e131c018458f">+96/-29</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>BUILD.md</strong><dd><code>New comprehensive binary build guide</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/Ajimaru/LM-Studio-Tray-Manager/pull/20/files#diff-00cbce1a205cdd2818c079f557aca61a04fcdd98b63e5cc98c729f5beae60d75">+382/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>python_docstrings.html</strong><dd><code>Document new functions and build module</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/Ajimaru/LM-Studio-Tray-Manager/pull/20/files#diff-6c3bc8abb33fc1da7bf3fe946170ba7a1b914b6f8773d4ed6cd06384a89d8f50">+159/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>SETUP.md</strong><dd><code>Document binary preference in autostart</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/Ajimaru/LM-Studio-Tray-Manager/pull/20/files#diff-22495e73839073b1b84e771b0b9097a8820e466365fd27b123f7fe7edeeb3047">+7/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>README.md</strong><dd><code>Add BUILD.md link and update structure</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/Ajimaru/LM-Studio-Tray-Manager/pull/20/files#diff-0b5ca119d2be595aa307d34512d9679e49186307ef94201e4b3dfa079aa89938">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>index.html</strong><dd><code>Reorder cards to feature BUILD.md guide</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/Ajimaru/LM-Studio-Tray-Manager/pull/20/files#diff-b04b38d4e36f7a7171aeb211bf933eaf36d41d9866ebbc3639f673f84dc350ae">+10/-5</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>README.md</strong><dd><code>Add binary release path and build instructions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/Ajimaru/LM-Studio-Tray-Manager/pull/20/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5">+90/-33</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>setup.sh</strong><dd><code>Add note about binary preference</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/Ajimaru/LM-Studio-Tray-Manager/pull/20/files#diff-4209d788ad32c40cbda3c66b3de47eefb929308ca703bb77a6382625986add17">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr></tbody></table>

</details>

___

## Release Notes - v0.3.0

* **New Features**
  * Standalone binary now available for faster deployment and startup.
  * Multiple installation paths: binary release, Python package, or advanced build.
  * Binary autostart integration for improved user experience.

* **Documentation**
  * New Binary Build Guide with comprehensive setup instructions.
  * Updated installation guides tailored to each deployment option.
  * Enhanced troubleshooting section with build logs reference.

* **Chores**
  * Version bumped to 0.3.0.

### CODE REVIEW

File: build_binary.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+    for src, dest in get_data_files():
+        cmd.extend(["--add-data", f"{src}{os.pathsep}{dest}"])
+
+    # Add GdkPixbuf loaders and cache
+    if loaders_dir and cache_file:
+        cmd.extend([
+            "--add-binary",
+            f"{loaders_dir}{os.pathsep}lib/gdk-pixbuf/loaders"
+        ])
+        cmd.extend([
```
[the remark this answers is not in the document]
Comment (Ajimaru) [reply build_binary.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/Ajimaru/LM-Studio-Tray-Manager/pull/20#discussion_r2836685479)

File: docs/BUILD.md [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
+      
+      - name: Install dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y upx gir1.2-gtk-3.0
+          pip install -r requirements.txt
+          pip install pyinstaller
```
[the remark this answers is not in the document]
Comment (Ajimaru) [reply docs/BUILD.md]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/Ajimaru/LM-Studio-Tray-Manager/pull/20#discussion_r2836790021)
